File: components/messaging-core/src/main/java/org/torquebox/messaging/core/Queue.java
package org.torquebox.messaging.core;


public class Queue extends Destination {
	
	public Queue() {
		
	}
	
	public void start() throws Exception {
		log.info( "start queue " + getName() );
		
		getServer().createQueue(false, getName(), "", false );
	}
	
	public void destroy() throws Exception {
		log.info( "destroy queue " + getName() );
		
		getServer().destroyQueue( getName() );
	}

}

File: gems/messaging/lib/torquebox/messaging/core/queue.rb
require 'torquebox/messaging/core/destination'

module TorqueBox
  module Messaging
    module Core
      class Queue < Destination
        def initialize(jms_queue)
          super
        end

        def to_s
          "[Queue: #{super}]"
        end

      end
    end
  end
end
